packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1062,6 +1113,49 @@ export function getClientInitialization(
   };
 }
 
+/**
+ * Get client initialization options for namespace or interface. The info is from `@clientInitialization` decorator.
+ *
+ * @param context
+ * @param entity namespace or interface which represents a client
+ * @returns
+ */
+export function getClientInitializationOptions(
+  context: TCGCContext,
+  entity: Namespace | Interface,
+): ClientInitializationOptions | undefined {
+  const options = getScopedDecoratorData(context, clientInitializationKey, entity);
+  if (options === undefined) return undefined;
+
+  // backward compatibility
+  if (
+    options.properties.get("initializedBy") === undefined &&
+    options.properties.get("parameters") === undefined
+  ) {
+    return {
+      parameters: options,
+    };
```

Comment:
No `initializedBy` value? I assume you need to provide one as well (either parent or individual, based on whether this is root client)

Reply:
the default logic is in `package.ts`, not included here.